diff --git a/configure b/configure
index ea00121..871e78b 100755
--- a/configure
+++ b/configure
@@ -10022,8 +10022,8 @@ if test "x$enable_ssp" != xno; then :
 
    save_CFLAGS="$CFLAGS"
    save_CXXFLAGS="$CXXFLAGS"
-   CXXFLAGS="$CXXFLAGS -Werror -fstack-protector-all -D_FORTIFY_SOURCE=2"
-   CFLAGS="$CFLAGS -Werror -fstack-protector-all -D_FORTIFY_SOURCE=2"
+   CXXFLAGS="$CXXFLAGS -Werror -fstack-protector-all"
+   CFLAGS="$CFLAGS -Werror -fstack-protector-all"
    cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int something ();
@@ -10032,8 +10032,8 @@ if ac_fn_c_try_compile "$LINENO"; then :
 
        { $as_echo "$as_me:${as_lineno-$LINENO}: Compiling with gcc -fstack-protector-all et al." >&5
 $as_echo "$as_me: Compiling with gcc -fstack-protector-all et al." >&6;}
-	CFLAGS="$save_CFLAGS -fstack-protector-all -D_FORTIFY_SOURCE=2"
-	CXXFLAGS="$save_CXXFLAGS -fstack-protector-all -D_FORTIFY_SOURCE=2"
+	CFLAGS="$save_CFLAGS -fstack-protector-all"
+	CXXFLAGS="$save_CXXFLAGS -fstack-protector-all"
 else
 
        { $as_echo "$as_me:${as_lineno-$LINENO}: Compiler does not support -fstack-protector-all et al." >&5
diff --git a/configure.ac b/configure.ac
index b4dd4af..9a885bc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -186,12 +186,12 @@ AC_ARG_ENABLE([ssp],
 AS_IF([test "x$enable_ssp" != xno],[
    save_CFLAGS="$CFLAGS"
    save_CXXFLAGS="$CXXFLAGS"
-   CXXFLAGS="$CXXFLAGS -Werror -fstack-protector-all -D_FORTIFY_SOURCE=2"
-   CFLAGS="$CFLAGS -Werror -fstack-protector-all -D_FORTIFY_SOURCE=2"
+   CXXFLAGS="$CXXFLAGS -Werror -fstack-protector-all"
+   CFLAGS="$CFLAGS -Werror -fstack-protector-all"
    AC_COMPILE_IFELSE([AC_LANG_SOURCE([int something ();])], [
        AC_MSG_NOTICE([Compiling with gcc -fstack-protector-all et al.])
-	CFLAGS="$save_CFLAGS -Werror -fstack-protector-all -D_FORTIFY_SOURCE=2"
-	CXXFLAGS="$save_CXXFLAGS -Werror -fstack-protector-all -D_FORTIFY_SOURCE=2"],[
+	CFLAGS="$save_CFLAGS -Werror -fstack-protector-all"
+	CXXFLAGS="$save_CXXFLAGS -Werror -fstack-protector-all"],[
        AC_MSG_NOTICE([Compiler does not support -fstack-protector-all et al.])
        CFLAGS="$save_CFLAGS"
        CXXFLAGS="$save_CXXFLAGS"])])
